pennylane/interfaces/tensorflow_autograph.py

```diff
@@ -236,3 +248,243 @@ def _backward(*all_params):
+            compute the gradients during the backwards pass.
+        gradient_kwargs (dict): dictionary of keyword arguments to pass when
+            determining the gradients of tapes
+        gradient_fn (callable): the gradient function to use to compute quantum gradients
+        _n (int): a positive integer used to track nesting of derivatives, for example
+            if the nth-order derivative is requested.
+        max_diff (int): If ``gradient_fn`` is a gradient transform, this option specifies
+            the maximum number of derivatives to support. Increasing this value allows
+            for higher order derivatives to be extracted, at the cost of additional
+            (classical) computational overhead during the backwards pass.
+        mode (str): Whether the gradients should be computed on the forward
+            pass (``forward``) or the backward pass (``backward``).
+
+    Returns:
+        list[list[tf.Tensor]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+    """
+    all_params = []
+    parameters = []
+    lens = []
+    trainable = []
+    output_types = []
+
+    for tape in tapes:
+        # store the trainable parameters
+        params = tape.get_parameters(trainable_only=False)
+        tape.trainable_params = qml.math.get_trainable_indices(params)
+
+        parameters += [p for i, p in enumerate(params) if i in tape.trainable_params]
+        all_params += params
+        trainable += (np.array(list(tape.trainable_params)) + sum(lens)).tolist()
+
+        lens.append(len(params))
+
+        if tape.all_sampled:
+            output_types.append(tf.int64)
+        elif tape.measurements[0].return_type is qml.measurements.State:
+            output_types.append(tf.complex128)
+        else:
+            output_types.append(tf.float64)
+
+    if mode == "forward":
+        output_types += [tf.float64] * len(tapes)
+
+    output_types += [tf.int32] * len(tapes)
+
+    def _nest_params(all_params):
+        count = 0
+        params_unwrapped = []
+
+        for s in lens:
+            params_unwrapped.append(all_params[count : count + s])
+            count += s
+
+        return params_unwrapped
+
+    def _forward(*all_params):
+        params_unwrapped = _nest_params(all_params)
+        output_sizes = []
+
+        with qml.tape.Unwrap(*tapes, params=params_unwrapped):
+            # Forward pass: execute the tapes
+            res, jacs = execute_fn(tapes, **gradient_kwargs)
+
+        for i, tape in enumerate(tapes):
+            # convert output to TensorFlow tensors
+            res[i] = tf.convert_to_tensor(res[i])
+
+            if tape.all_sampled:
+                res[i] = tf.cast(res[i], tf.int64)
+
+            output_sizes.append(tf.size(res[i]))
+
+        if jacs:
+            for i, jac in enumerate(jacs):
+                jacs[i] = tf.convert_to_tensor(jac)

```

Comment:
No, because due to limitations with TensorFlow, this function needs to return a tuple of tensors. If we don't do any postprocessing here it would return a deeper tuple structure.

Reply:
Thanks!